src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -0,0 +1,376 @@
+        """
+        return TextGenerationInput
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        """
+        Property to return the output schema for the pipeline.
+
+        :return: the output schema for the pipeline
+        """
+        return TextGenerationOutput
+
+    def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
+        """
+        Convert the input schema for the pipeline to the inputs for the engine.
+
+        :param inputs: the input schema for the pipeline
+        :return: the inputs for the engine
+        """
+
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+
+        input_tokens = self.tokenizer(
+            inputs.sequences,
+            return_tensors="np",
+            max_length=self.sequence_length,
+            padding="max_length",
+        )
+
+        attention_mask = input_tokens["attention_mask"]
+
+        # TODO: Positions input is not required by BLOOM
+        # let's make it optional in the future
+        positions = attention_mask.cumsum(1) * attention_mask
+        positions -= 1  # assert that positions start at 0
+        positions_input = dict(positions=positions)
+
+        input_tokens = {**input_tokens, **positions_input}
+        onnx_input_names = self.multitoken_engine.onnx_input_names_no_cache
+        engine_input = self.tokens_to_engine_input(input_tokens, onnx_input_names)
+
+        if inputs.session_id is not None:
+            # if session_id is provided, we need to set it in engines
+            self.engine.session_id = inputs.session_id
+            self.multitoken_engine.session_id = inputs.session_id
+
+        postprocessing_kwargs = dict(return_logits=inputs.return_logits)
+        return engine_input, postprocessing_kwargs
+
+    def process_engine_outputs(
+        self, engine_outputs: List[numpy.ndarray], **kwargs
+    ) -> TextGenerationOutput:
+        """
+        Convert the engine outputs to the output schema for the pipeline.
+
+        :param engine_outputs: the outputs from the engine
+        :return: the output schema for the pipeline
+        """
+        generated_tokens, generated_logits = engine_outputs
+        sequences = self.tokenizer.batch_decode(
+            *generated_tokens, skip_special_tokens=True
+        )
+        logits = generated_logits if kwargs.get("return_logits") else None
+
+        return TextGenerationOutput(sequences=sequences, logits=logits)
+
+    def engine_forward(
+        self, engine_inputs: List[numpy.ndarray], **kwargs
+    ) -> Tuple[numpy.ndarray, numpy.ndarray]:
+        """
+        Run the forward pass on the engine.
+
+        :param engine_inputs: list of numpy inputs to
+            Pipeline engine forward pass
+        :return: A tuple of numpy array that contains the
+            sequence of generated tokens and a sequence
+            of logits for each generated token
+        """
+        if not self.multitoken_engine.kv_cache_enabled:
```

Comment:
same here, we should be just use max generated tokens

Reply:
if not self.multitoken_engine.kv_cache_enabled and self.max_generated_tokens != 1: error